csharp/src/Ice/RouterInfo.cs

```diff
@@ -110,32 +110,9 @@ internal async ValueTask<IReadOnlyList<Endpoint>> GetClientEndpointsAsync(Cancel
             {
                 if (_clientEndpoints == null)
                 {
+                    // If GetClientProxy() returns nil, use router endpoints.
+                    _clientEndpoints = (clientProxy ?? Router).IceReference.Endpoints;
```

Comment:
We no longer clear the Router from client proxy

Reply:
Yes, it's not longer necessary as we no longer need to obtain the connection from the client proxy.